src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py

```diff
@@ -119,17 +120,25 @@ def parse_issues(self, clang_output, revision):
         '''
         Parse clang-tidy output into structured issues
         '''
-
-        # Limit clang output parsing to 'Enabled checks:'
-        end = re.search(r'^Enabled checks:\n', clang_output, re.MULTILINE)
-        if end is not None:
-            clang_output = clang_output[:end.start()-1]
+        # Detect end of file warnings count
+        # When an invalid file is used, this line does not appear
+        has_warnings = REGEX_HAS_WARNINGS.search(clang_output)
+        if has_warnings is None:
```

Comment:
Could you add tests for these two new cases? (empty output and no issues found)

Reply:
Not really, because i use clang-tidy directly in the unit test, as `mach` is not available .
`clang-tidy` does not have the line in its output, i append it in `conftest.py`

We **really** need integration tests.

Reply:
You could write a unit test that calls `parse_issues` directly, with a "fake" clang_output based on what you've seen with mach.

Reply:
Jus to trigger the exception (or the analysis) ? Sure

Reply:
To trigger the exception, and to test these two cases (no output, 0 issues)